File: source/Chargeable Axe and Pickaxe/Common/Integrations/GenericModConfigMenu/IGenericModConfigMenuApi.cs
/*************************************************
**
** You're viewing a file in the SMAPI mod dump, which contains a copy of every open-source SMAPI mod
** for queries and analysis.
**
** This is *not* the original file, and not necessarily the latest version.
** Source repository: https://gitlab.com/theLion/smapi-mods
**
*************************************************/

using StardewModdingAPI;
using System;

namespace TheLion.Common.Integrations.GenericModConfigMenu
{
	/// <summary>The API provided by the Generic Mod Config Menu mod.</summary>
	public interface IGenericModConfigMenuApi
	{
		void RegisterModConfig(IManifest mod, Action revertToDefault, Action saveToFile);
		void RegisterLabel(IManifest mod, string labelName, string labelDesc);
		void RegisterSimpleOption(IManifest mod, string optionName, string optionDesc, Func<bool> optionGet, Action<bool> optionSet);
		void RegisterSimpleOption(IManifest mod, string optionName, string optionDesc, Func<SButton> optionGet, Action<SButton> optionSet);
		void RegisterSimpleOption(IManifest mod, string optionName, string optionDesc, Func<string> optionGet, Action<string> optionSet);
		void RegisterChoiceOption(IManifest mod, string optionName, string optionDesc, Func<string> optionGet, Action<string> optionSet, string[] choices);
		void RegisterClampedOption(IManifest mod, string optionName, string optionDesc, Func<int> optionGet, Action<int> optionSet, int min, int max);
		void RegisterClampedOption(IManifest mod, string optionName, string optionDesc, Func<float> optionGet, Action<float> optionSet, float min, float max);
	}
}

File: source/P2P Animal Noise/IGenericModConfigMenuApi.fs
namespace GenericModConfigMenu

open System
open StardewModdingAPI

type IGenericModConfigMenuApi =
    abstract member Register: IManifest * Action * Action * bool -> unit

    abstract member AddKeybind:
        IManifest * Func<SButton> * Action<SButton> * Func<string> * Func<string> * string -> unit

    abstract member AddTextOption:
        IManifest *
        Func<string> *
        Action<string> *
        Func<string> *
        Func<string> *
        string array *
        Func<string, string> *
        string ->
            unit
